File: assignments/haskell/roman-numerals/roman-numerals_test.hs
import Test.HUnit (Assertion, (@=?), runTestTT, Test(..))
import Control.Monad (void)
import Roman (numerals)

testCase :: String -> Assertion -> Test
testCase label assertion = TestLabel label (TestCase assertion)

main :: IO ()
main = void $ runTestTT $ TestList
       [ TestList numeralsTests ]

numeralsTests :: [Test]
numeralsTests = map TestCase $
  [ "I" @=? numerals 1
  , "II" @=? numerals 2
  , "III" @=? numerals 3
  , "IV" @=? numerals 4
  , "V" @=? numerals 5
  , "VI" @=? numerals 6
  , "IX" @=? numerals 9
  , "XXVII" @=? numerals 27
  , "XLVIII" @=? numerals 48
  , "LIX" @=? numerals 59
  , "XCIII" @=? numerals 93
  , "CXLI" @=? numerals 141
  , "CLXIII" @=? numerals 163
  , "CDII" @=? numerals 402
  , "DLXXV" @=? numerals 575
  , "CMXI" @=? numerals 911
  , "MXXIV" @=? numerals 1024
  , "MMM" @=? numerals 3000
  ]

File: assignments/clojure/roman-numerals/roman_numerals_test.clj
(ns roman.test (:require [clojure.test :refer :all]))
(load-file "roman.clj")

(deftest one
  (is (= "I" (roman/numerals 1))))

(deftest two
  (is (= "II" (roman/numerals 2))))

(deftest three
  (is (= "III" (roman/numerals 3))))

(deftest four
  (is (= "IV" (roman/numerals 4))))

(deftest five
  (is (= "V" (roman/numerals 5))))

(deftest six
  (is (= "VI" (roman/numerals 6))))

(deftest nine
  (is (= "IX" (roman/numerals 9))))

(deftest twenty-seven
  (is (= "XXVII" (roman/numerals 27))))

(deftest forty-eight
  (is (= "XLVIII" (roman/numerals 48))))

(deftest fifty-nine
  (is (= "LIX" (roman/numerals 59))))

(deftest ninety-three
  (is (= "XCIII" (roman/numerals 93))))

(deftest one-hundred-forty-one
  (is (= "CXLI" (roman/numerals 141))))

(deftest one-hundred-sixty-three
  (is (= "CLXIII" (roman/numerals 163))))

(deftest four-hundred-two
  (is (= "CDII" (roman/numerals 402))))

(deftest five-hundred-seventy-five
  (is (= "DLXXV" (roman/numerals 575))))

(deftest nine-hundred-eleven
  (is (= "CMXI" (roman/numerals 911))))

(deftest one-thousand-twenty-four
  (is (= "MXXIV" (roman/numerals 1024))))

(deftest three-thousand
  (is (= "MMM" (roman/numerals 3000))))

(run-tests)
